compiler/cbmc/src/cbmc_string.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+use string_interner::symbol::SymbolU32;
+use string_interner::StringInterner;
+
+/// This class implements an interner for Strings.
+/// CBMC objects to have a large number of strings which refer to names: symbols, files, etc.
+/// These tend to be reused many times, which causes signifcant memory usage.
```

Comment:
Instead of/along with "significant", maybe give ballpark reduction?

Reply:
On the stdlib test, this saves 146MB on a 1GB initial.  It used to save more, before we started streaming the ireps.

Reply:
My suggestion would be to put a rough percentage in the comment, so if someone is looking at this without seeing the PR itself they have that context!

Reply:
done